src/utils/get-command-spec.ts

```diff
@@ -23,6 +23,79 @@ export type EmptyCommandInput = {
 	[Key in keyof Required<CommandInput>]: undefined;
 };
 
+// Determine whether type is a literal or not
+type IsLiteral<T> = T extends Array<infer Item>
+	? IsLiteral<Item>
+	: string extends T
+	? false
+	: number extends T
+	? false
+	: true;
+
+// Hand back array of provided type, whether it was an array or not
+type AlwaysArray<T> = NonNullable<T> extends Array<unknown> ? T : NonNullable<T>[];
+
+// Create union of all basic types present in the type
+type UnionTypes<T> = T extends Array<infer Item>
+	? `arrayOf${UnionTypes<Item>}`
+	: T extends string
+	? IsLiteral<T> extends true
+		? 'strvalue'
+		: 'string'
+	: T extends number
+	? IsLiteral<T> extends true
+		? 'numvalue'
+		: 'number'
+	: T extends boolean
+	? 'boolean'
+	: T extends Function // eslint-disable-line
+	? 'function'
+	: 'object';
+
+// Determine whether it's a union or not
+type IsUnion<T, U extends T = T> = T extends unknown ? ([U] extends [T] ? false : true) : false;
+
+// Decide whether the type contains multiple base types
+type DetectMultipleTypes<T> = IsUnion<UnionTypes<T>>;
+
+// Decide what the accepts type will expand to
+type AcceptTypes<T> = IsLiteral<T> extends true
+	? AlwaysArray<T>
+	: AlwaysArray<T> | (() => AlwaysArray<T> | Promise<AlwaysArray<T>>);
+
+// Get type of array elements
+type ArrayItemType<T> = T extends Array<infer Item> ? Item : T;
+
+// Decide whether accepts should be optional or not
+type IsAcceptsOptional<T> = IsLiteral<T> extends true ? false : true;
+
+// Decide whether numeric type identifier is needed; decide whether to flag as required; decide whether multiple responses are accepted; decide whether accepts is optional or not, and determine type for accepts
+type DataTypeAddons<T, U> = {
+	/** What type of numeric data should be provided. Invalid data will be rejected. */
+	type: number extends U ? (U extends number ? 'integer' | 'float' : ExcludeMe) : ExcludeMe;
+
+	/** Must be defined as true when data must be provided to this command, otherwise must be omitted. */
+	required: undefined extends U ? ExcludeMe : true;
+
+	/** Must be defined as true when multiple elements are accepted, otherwise must be omitted. */
```

Comment:
This should be wrapped in backticks to be formatted properly:

```suggestion
	/** Must be defined as `true` when multiple elements are accepted, otherwise must be omitted. */
```